File: ProyectoSteam/app/src/main/java/com/example/myapplication/Formulario.java
package com.example.myapplication;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import javax.mail.Message;
import javax.mail.MessagingException;
import javax.mail.PasswordAuthentication;
import javax.mail.Session;
import javax.mail.Transport;
import javax.mail.internet.InternetAddress;
import javax.mail.internet.MimeMessage;
import java.util.Properties;

public class Formulario extends AppCompatActivity {

    private EditText editTextNombre;
    private EditText editTextCorreo;
    private CheckBox checkBoxMayorEdad;

    private static final String SMTP_HOST = "smtp.gmail.com";
    private static final String SMTP_PORT = "587";
    private static final String EMAIL = "tu correo electronico";
    private static final String PASSWORD = "contraseña smtp";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_formulario);

        editTextNombre = findViewById(R.id.editTextNombre);
        editTextCorreo = findViewById(R.id.editTextCorreo);
        checkBoxMayorEdad = findViewById(R.id.checkBoxMayorEdad);

        Button botonVolver = findViewById(R.id.buttonVolver);
        botonVolver.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                volverAlMainActivity(v);
            }
        });
    }

    public void enviarFormulario(View view) {
        String nombre = editTextNombre.getText().toString();
        String correo = editTextCorreo.getText().toString();
        boolean esMayorEdad = checkBoxMayorEdad.isChecked();

        String subject = "Miniproyecto";
        String body = "Para mas proyectos sigueme en: https://github.com/marccordero";

        new SendEmailTask().execute(subject, body, correo);
    }

    private class SendEmailTask extends AsyncTask<String, Void, Boolean> {

        @Override
        protected Boolean doInBackground(String... params) {
            String subject = params[0];
            String body = params[1];
            String recipient = params[2];

            try {
                sendEmail(subject, body, recipient);
                return true;
            } catch (MessagingException e) {
                e.printStackTrace();
                return false;
            }
        }

        @Override
        protected void onPostExecute(Boolean success) {
            if (success) {
                Toast.makeText(Formulario.this, "Correo enviado correctamente", Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(Formulario.this, "Error al enviar el correo", Toast.LENGTH_SHORT).show();
            }


            Intent intent = new Intent(Formulario.this, MainActivity.class);
            startActivity(intent);
            finish();
        }
    }


    private void sendEmail(String subject, String body, String recipient) throws MessagingException {
        Properties properties = new Properties();
        properties.put("mail.smtp.auth", "true");
        properties.put("mail.smtp.starttls.enable", "true");
        properties.put("mail.smtp.host", SMTP_HOST);
        properties.put("mail.smtp.port", SMTP_PORT);

        Session session = Session.getInstance(properties, new javax.mail.Authenticator() {
            protected PasswordAuthentication getPasswordAuthentication() {
                return new PasswordAuthentication(EMAIL, PASSWORD);
            }
        });

        Message message = new MimeMessage(session);
        message.setFrom(new InternetAddress(EMAIL));
        message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(recipient));
        message.setSubject(subject);
        message.setText(body);

        Transport.send(message);
    }
    public void volverAlMainActivity(View view) {
        Intent intent = new Intent(Formulario.this, MainActivity.class);
        startActivity(intent);
        finish();
    }


}

File: ProyectoSteam/app/src/main/java/com/example/myapplication/Formulario.kt
package com.example.myapplication

import androidx.appcompat.app.AlertDialog
import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import javax.mail.Message
import javax.mail.MessagingException
import javax.mail.PasswordAuthentication
import javax.mail.Session
import javax.mail.Transport
import javax.mail.internet.InternetAddress
import javax.mail.internet.MimeMessage
import java.util.*

class Formulario : AppCompatActivity() {

    private lateinit var editTextNombre: EditText
    private lateinit var editTextCorreo: EditText
    private lateinit var checkBoxMayorEdad: CheckBox

    private val SMTP_HOST = "smtp.gmail.com"
    private val SMTP_PORT = "587"
    private val EMAIL = "marccordero003@gmail.com"
    private val PASSWORD = "fznttxvntpqgwcuc"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_formulario)

        editTextNombre = findViewById(R.id.editTextNombre)
        editTextCorreo = findViewById(R.id.editTextCorreo)
        checkBoxMayorEdad = findViewById(R.id.checkBoxMayorEdad)

        val botonVolver: Button = findViewById(R.id.buttonVolver)
        botonVolver.setOnClickListener {
            volverAlMainActivity(it)
        }
    }

    fun enviarFormulario(view: View) {
        val nombre = editTextNombre.text.toString()
        val correo = editTextCorreo.text.toString()
        val esMayorEdad = checkBoxMayorEdad.isChecked

        if (nombre.isEmpty() || correo.isEmpty() || !esMayorEdad) {
            // Mostrar AlertDialog indicando campos faltantes
            mostrarAlertaCamposFaltantes()
        } else {
            val subject = "Miniproyecto"
            val body = "Para mas proyectos sigueme en: https://github.com/marccordero"

            SendEmailTask().execute(subject, body, correo)
        }
    }

    private inner class SendEmailTask : AsyncTask<String, Void, Boolean>() {
        override fun doInBackground(vararg params: String): Boolean {
            val subject = params[0]
            val body = params[1]
            val recipient = params[2]

            return try {
                sendEmail(subject, body, recipient)
                true
            } catch (e: MessagingException) {
                e.printStackTrace()
                false
            }
        }

        override fun onPostExecute(success: Boolean) {
            if (success) {
                Toast.makeText(this@Formulario, "Correo enviado correctamente", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this@Formulario, "Error al enviar el correo", Toast.LENGTH_SHORT).show()
            }

            val intent = Intent(this@Formulario, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    @Throws(MessagingException::class)
    private fun sendEmail(subject: String, body: String, recipient: String) {
        val properties = Properties()
        properties["mail.smtp.auth"] = "true"
        properties["mail.smtp.starttls.enable"] = "true"
        properties["mail.smtp.host"] = SMTP_HOST
        properties["mail.smtp.port"] = SMTP_PORT

        val session = Session.getInstance(properties, object : javax.mail.Authenticator() {
            override fun getPasswordAuthentication(): PasswordAuthentication {
                return PasswordAuthentication(EMAIL, PASSWORD)
            }
        })

        val message = MimeMessage(session)
        message.setFrom(InternetAddress(EMAIL))
        message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(recipient))
        message.subject = subject
        message.setText(body)

        Transport.send(message)
    }

    fun volverAlMainActivity(view: View) {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
    private fun mostrarAlertaCamposFaltantes() {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Campos faltantes")
        builder.setMessage("Por favor completa todos los campos del formulario.")
        builder.setPositiveButton("OK") { dialog, _ ->
            dialog.dismiss()
        }
        val dialog = builder.create()
        dialog.show()
    }
}
